test/vcf/optional_policy_test.cpp

```diff
@@ -0,0 +1,140 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "catch/catch.hpp"
+
+#include "vcf/error.hpp"
+#include "vcf/optional_policy.hpp"
+
+namespace ebi
+{
+    
+    TEST_CASE("Info predefined tags warnings", "[body info warnings]")
+    {            
+        std::shared_ptr<vcf::Source> source{
+            new vcf::Source{
+                "Example VCF source",
+                vcf::InputFormat::VCF_FILE_VCF | vcf::InputFormat::VCF_FILE_BGZIP,
+                vcf::Version::v43,
+                vcf::Ploidy{2, {{"Y", 1}}},
+                {},
+                { "Sample1" }}};
+
+        source->meta_entries.emplace(vcf::REFERENCE,
+            vcf::MetaEntry{
+                1,
+                vcf::REFERENCE,
+                "file",
+                source
+        });
+
+        source->meta_entries.emplace(vcf::CONTIG,
+            vcf::MetaEntry{
+                1,
+                vcf::CONTIG,
+                { { vcf::ID, "chr1" } },
+                source
+        });
+
+        source->meta_entries.emplace(vcf::FORMAT,
+            vcf::MetaEntry{
+                1,
+                vcf::FORMAT,
+                {
+                    { vcf::ID, vcf::GT },
+                    { vcf::NUMBER, "1" },
+                    { vcf::TYPE, vcf::STRING },
+                    { vcf::DESCRIPTION, "Genotype" }
+                },
+                source
+        });
+
+        vcf::ParsingState parsing_state{source};
+
+        vcf::ValidateOptionalPolicy optional_policy;
+
+        SECTION("Confidence Interval Tags") 
+        {
+            std::vector<std::string> confidence_interval_tags = { 
+                                                                    vcf::CICN,
+                                                                    vcf::CICNADJ,
+                                                                    vcf::CIEND,
+                                                                    vcf::CILEN,
+                                                                    vcf::CIPOS
+                                                                };
+
+            for (auto & confidence_interval_tag : confidence_interval_tags) {
+
+                source->meta_entries.emplace(vcf::INFO,
+                    vcf::MetaEntry{
+                        1,
+                        vcf::INFO,
+                        {
+                            { vcf::ID, confidence_interval_tag },
+                            { vcf::NUMBER, "2" },
+                            { vcf::TYPE, vcf::INTEGER },
+                            { vcf::DESCRIPTION, "CI tag" }
+                        },
+                        source
+                });
+
+                CHECK_NOTHROW( (optional_policy.optional_check_body_entry(parsing_state, vcf::Record{
```

Comment:
does this have to be inside the `for`?

Reply:
I used the `for` to avoid rewriting the same test and adding similar meta entries again, for each of those 5 confidence interval tags.

I'm sorry I couldn't understand clearly, are you asking about removing the `CHECK_NOTHROW`?

Reply:
oh, I see, I got confused, the code is fine